src/main.dart

```diff
@@ -1,8 +1,107 @@
-// TODO: Run task list app
+
+import 'dart:io';
+import 'task_list.dart';
+
+ TaskList tasks = TaskList();
 void main(List<String> args) {
-  print('Hello World');
+  printMenu();
+
+  bool choiceValid = false;
+  int choice = -1;
+  while(!choiceValid) {
+    try {
+      stdout.write("Enter your choice: ");
+      choice = int.parse(stdin.readLineSync()!);
+      choiceValid = true;
+  } catch(e) {
+    print("Please enter a number");
+  }
+}
+switch(choice) {
+  case 1:
+  addTask();
+  break;
+  case 2:
+  editTask();
+  break;
+  case 3:
+  deleteTask();
+  break;
+  case 4:
+  markTaskAsCompleted();
+  break;
+  case 5:
+  tasks.printTasks();
+  break;
+  case 6:
+  print("Thank you for using the Task List App");
+  default:
+  print("Invalid choice");
+  break;
+
+
+}
```

Comment:
whole thing should be in a loop. The program shuts down immediately making me unable to test it